drivers/mipi_dsi/dsi_stm32.c

```diff
@@ -0,0 +1,465 @@
+	LOG_DBG("  VerticalFrontPorch %u", data->vid_cfg.VerticalFrontPorch);
+	LOG_DBG("  VerticalActive %u", data->vid_cfg.VerticalActive);
+	LOG_DBG("  LPCommandEnable 0x%x", data->vid_cfg.LPCommandEnable);
+	LOG_DBG("  LPLargestPacketSize %u", data->vid_cfg.LPLargestPacketSize);
+	LOG_DBG("  LPVACTLargestPacketSize %u", data->vid_cfg.LPVACTLargestPacketSize);
+	LOG_DBG("  LPHorizontalFrontPorchEnable 0x%x", data->vid_cfg.LPHorizontalFrontPorchEnable);
+	LOG_DBG("  LPHorizontalBackPorchEnable 0x%x", data->vid_cfg.LPHorizontalBackPorchEnable);
+	LOG_DBG("  LPVerticalActiveEnable 0x%x", data->vid_cfg.LPVerticalActiveEnable);
+	LOG_DBG("  LPVerticalFrontPorchEnable 0x%x", data->vid_cfg.LPVerticalFrontPorchEnable);
+	LOG_DBG("  LPVerticalBackPorchEnable 0x%x", data->vid_cfg.LPVerticalBackPorchEnable);
+	LOG_DBG("  LPVerticalSyncActiveEnable 0x%x", data->vid_cfg.LPVerticalSyncActiveEnable);
+	LOG_DBG("  FrameBTAAcknowledgeEnable 0x%x", data->vid_cfg.FrameBTAAcknowledgeEnable);
+
+	if (IS_ENABLED(CONFIG_MIPI_DSI_STM32_HOST_TIMEOUTS)) {
+		DSI_HOST_TimeoutTypeDef *ht = &data->host_timeouts;
+
+		LOG_DBG("HAL_DSI_ConfigHostTimeouts:");
+		LOG_DBG("  TimeoutCkdiv %u", ht->TimeoutCkdiv);
+		LOG_DBG("  HighSpeedTransmissionTimeout %u", ht->HighSpeedTransmissionTimeout);
+		LOG_DBG("  LowPowerReceptionTimeout %u", ht->LowPowerReceptionTimeout);
+		LOG_DBG("  HighSpeedReadTimeout %u", ht->HighSpeedReadTimeout);
+		LOG_DBG("  LowPowerReadTimeout %u", ht->LowPowerReadTimeout);
+		LOG_DBG("  HighSpeedWriteTimeout %u", ht->HighSpeedWriteTimeout);
+		LOG_DBG("  HighSpeedWritePrespMode %u", ht->HighSpeedWritePrespMode);
+		LOG_DBG("  LowPowerWriteTimeout %u", ht->LowPowerWriteTimeout);
+		LOG_DBG("  BTATimeout %u", ht->BTATimeout);
+	}
+
+	if (IS_ENABLED(CONFIG_MIPI_DSI_STM32_PHY_TIMINGS)) {
+		LOG_DBG("HAL_DSI_ConfigPhyTimer:");
+		LOG_DBG("  ClockLaneHS2LPTime %u", data->phy_timings.ClockLaneHS2LPTime);
+		LOG_DBG("  ClockLaneLP2HSTime %u", data->phy_timings.ClockLaneLP2HSTime);
+		LOG_DBG("  DataLaneHS2LPTime %u", data->phy_timings.DataLaneHS2LPTime);
+		LOG_DBG("  DataLaneLP2HSTime %u", data->phy_timings.DataLaneLP2HSTime);
+		LOG_DBG("  DataLaneMaxReadTime %u", data->phy_timings.DataLaneMaxReadTime);
+		LOG_DBG("  StopWaitTime %u", data->phy_timings.StopWaitTime);
+	}
+}
+
+static int mipi_dsi_stm32_host_init(const struct device *dev)
+{
+	const struct mipi_dsi_stm32_config *config = dev->config;
+	struct mipi_dsi_stm32_data *data = dev->data;
+	struct stm32_pclken pclken_tmp;
+	uint32_t hse_clock;
+	int ret;
+
+	switch (config->data_lanes) {
+	case 1:
+		data->hdsi.Init.NumberOfLanes = DSI_ONE_DATA_LANE;
+		break;
+	case 2:
+		data->hdsi.Init.NumberOfLanes = DSI_TWO_DATA_LANES;
+		break;
+	default:
+		LOG_ERR("Number of DSI lanes (%d) not supported!", config->data_lanes);
+		return -ENOTSUP;
+	}
+
+	pclken_tmp.bus = STM32_SRC_PLL3_R;
+	ret = clock_control_get_rate(config->rcc, &pclken_tmp, &data->pixel_clk_khz);
```

Comment:
why doesn't mipi node have a `clocks` node, so you can get that from DT?

Reply:
Could you suggest the way to go to get the PLL3_R and crystal clock?
I have following clocks entry:
```dts
mipi_dsi: dsihost@50000000 {
	clocks = <&rcc STM32_CLOCK_BUS_APB3 0x00000010>;
};
```

Reply:
You can add domain clocks:
```
mipi_dsi: dsihost@50000000 {
	clocks = <&rcc STM32_CLOCK_BUS_APB3 0x00000010>,
                      <&rcc STM32_SRC_PLL3_R DSI_SEL(1)>;
};
```